From a84c8ddb19beef7580d6a4b8c417bb188517412a Mon Sep 17 00:00:00 2001
From: Karol Herbst <kherbst@redhat.com>
Date: Thu, 4 Mar 2021 20:32:10 +0100
Subject: [PATCH 1/2] tegra/context: fix regression in tegra_draw_vbo

We should only pass in a new indirect_info object if we actually set valid
values in it.

Fixes: abe8ef862fe5 "gallium: make pipe_draw_indirect_info * a draw_vbo parameter"
Signed-off-by: Karol Herbst <kherbst@redhat.com>
Reviewed-by: Ilia Mirkin <imirkin@alum.mit.edu>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/9425>
---
 src/gallium/drivers/tegra/tegra_context.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gallium/drivers/tegra/tegra_context.c b/src/gallium/drivers/tegra/tegra_context.c
index 1d976267aee..97b45ed1627 100644
--- a/src/gallium/drivers/tegra/tegra_context.c
+++ b/src/gallium/drivers/tegra/tegra_context.c
@@ -75,13 +75,13 @@ tegra_draw_vbo(struct pipe_context *pcontext,
       if (pindirect && pindirect->buffer) {
          memcpy(&indirect, pindirect, sizeof(indirect));
          indirect.buffer = tegra_resource_unwrap(pindirect->buffer);
+         pindirect = &indirect;
       }
 
       if (pinfo->index_size && !pinfo->has_user_indices)
          info.index.resource = tegra_resource_unwrap(info.index.resource);
 
       pinfo = &info;
-      pindirect = &indirect;
    }
 
    context->gpu->draw_vbo(context->gpu, pinfo, pindirect, draws, num_draws);
-- 
2.30.1

